packages/create-waku/src/index.ts

```diff
@@ -129,55 +184,116 @@ async function init() {
-    name: packageName ?? toValidPackageName(targetDir),
-    version: '0.0.0',
-  };
+      repoInfo = await getRepoInfo(repoUrl);
 
-  console.log('Setting up project...');
+      // NOTE validate reproInfo
+      if (!repoInfo) {
+        console.error(
+          `Found invalid GitHub URL: ${red(
+            `"${example}"`,
+          )}. Please fix the URL and try again.`,
+        );
+        process.exit(1);
+      }
 
-  const templateDir = path.join(templateRoot, chooseTemplate || CHOICES[0]!);
+      const found = await hasRepo(repoInfo);
+      // NOTE Do the repo exist?
+      if (!found) {
+        console.error(
+          `Could not locate the repository for ${red(
+            `"${example}"`,
+          )}. Please check that the repository exists and try again.`,
+        );
+        process.exit(1);
+      }
+    } else {
+      const found = await existsInRepo(example);
 
-  // Read existing package.json from the root directory
-  const packageJsonPath = path.join(root, 'package.json');
+      if (!found) {
+        console.error(
+          `Could not locate an example named ${red(
+            `"${example}"`,
+          )}. Please check that the example exists and try again.`,
+        );
+        process.exit(1);
+      }
+    }
+  }
 
-  // Read new package.json from the template directory
-  const newPackageJsonPath = path.join(templateDir, 'package.json');
-  const newPackageJson = JSON.parse(
-    await fsPromises.readFile(newPackageJsonPath, 'utf-8'),
-  );
+  console.log('Setting up project...');
 
-  fse.copySync(templateDir, root);
+  const root = path.resolve(targetDir);
+  const { packageName, shouldOverwrite } = result;
 
-  if (existsSync(path.join(root, 'gitignore'))) {
-    await fsPromises.rename(
-      path.join(root, 'gitignore'),
-      path.join(root, '.gitignore'),
-    );
+  if (shouldOverwrite) {
+    fse.emptyDirSync(root);
+  } else if (!existsSync(root)) {
+    await fsPromises.mkdir(root, { recursive: true });
   }
 
-  await fsPromises.writeFile(
-    packageJsonPath,
-    JSON.stringify(
-      {
-        ...newPackageJson,
-        ...pkg,
-      },
-      null,
-      2,
-    ),
-  );
+  if (example) {
+    /**
+     * If an example repository is provided, clone it.
+     */
+    try {
+      if (repoInfo) {
+        console.log(
+          `Downloading files from repo ${cyan(example)}. This might take a moment.`,
+        );
+        console.log();
+        await downloadAndExtractRepo(root, repoInfo);
+      } else {
+        console.log(
+          `Downloading files for example ${cyan(example)}. This might take a moment.`,
+        );
+        console.log();
+        await downloadAndExtractExample(root, example);
+      }
+    } catch (reason) {
+      // download error
+      throw new Error(isErrorLike(reason) ? reason.message : reason + '');
+    }
+
+    // TODO automatically installing dependencies
+    // 1. check packageManager
+    // 2. and then install dependencies
```

Comment:
can we use this? https://github.com/antfu/ni

Reply:
Oh thanks.
My plan is just below:

```
exec(`${packageManager} install`)
```

i don't know [ni](https://github.com/antfu/ni) but if it is useful, let's use it.

Reply:
Yes that can work too. just wanted to drop a possibility.

Reply:
Pay special attention when choosing a dependency. we generally avoid adding new dependencies especially if they are new.